server/src/main/java/org/opensearch/action/search/SearchTransportService.java

```diff
@@ -241,16 +244,30 @@ public void sendExecuteQuery(
         // we optimize this and expect a QueryFetchSearchResult if we only have a single shard in the search request
         // this used to be the QUERY_AND_FETCH which doesn't exist anymore.
         final boolean fetchDocuments = request.numberOfShards() == 1;
-        Writeable.Reader<SearchPhaseResult> reader = fetchDocuments ? QueryFetchSearchResult::new : QuerySearchResult::new;
+        // System.setProperty(FeatureFlags.PROTOBUF, "true");
+        if (FeatureFlags.isEnabled(FeatureFlags.PROTOBUF_SETTING)) {
+            ProtobufWriteable.Reader<SearchPhaseResult> reader = fetchDocuments ? QueryFetchSearchResult::new : QuerySearchResult::new;
+
+            final ActionListener handler = responseWrapper.apply(connection, listener);
+            transportService.sendChildRequest(
+                connection,
+                QUERY_ACTION_NAME,
+                request,
+                task,
+                new ProtobufConnectionCountingHandler<>(handler, reader, clientConnections, connection.getNode().getId())
+            );
+        } else {
```

Comment:
The handler class is different but the rest is the same. Collapse the similar parts.